File: perl/roberto.pl
use Time::HiRes qw(time);

my $start = time;

my @letters = split(//, 'ABCDEFGHIJKLMNOPQRSTUVWXYZ');
my @ar = ();

for(my $i = 0; $i < 10000; $i++) {
	my $size = scalar(@letters);
	push @ar, @letters[int(rand($size))] . @letters[int(rand($size))] . @letters[int(rand($size))] . @letters[int(rand($size))];
}

@ar = sort @ar;

my $output = '';

for(my $i = 0; $i < scalar(@ar); $i++) {
	$output .= "@ar[$i]\n";
}

open(my $fh, '>', 'roberto_output.txt');
print $fh $output;
close $fh;

print "Elapsed time: ", (time - $start) * 1000 ,"ms\n";

File: python/roberto.py
import datetime
import random

start = datetime.datetime.now()
letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'

ar = []

for i in range(0,10000):
    ar.append(letters[random.randint(0,len(letters) - 1)] + letters[random.randint(0,len(letters)-1)] + letters[random.randint(0,len(letters)-1)] + letters[random.randint(0,len(letters)-1)])

ar.sort();

output = ''

for i in range(0,10000):
    output += ar[i] + '\n'

f = open('roberto_output.txt', 'w')
f.write(output)
f.close()

end = datetime.datetime.now() - start

print 'Elapsed time: ' + str(end.days*86400000 + end.seconds*1000 + end.microseconds/1000) + 'ms\n'
